Scarab/ViewModels/ModListViewModel.cs

```diff
@@ -445,12 +445,26 @@ private async Task OnEnable(ModItem item)
             {
                 var dependents = _reverseDependencySearch.GetAllEnabledDependents(item).ToList();
 
-                if (!item.EnabledIsChecked ||
-                    dependents.Count == 0 ||
-                    await DisplayErrors.DisplayHasDependentsWarning(item.Name, dependents))
+                if (item.EnabledIsChecked && 
+                    (dependents.Count == 0 || 
+                    await DisplayErrors.DisplayHasDependentsWarning(item.Name, dependents)))
                 {
                     await _installer.Toggle(item);
                 }
+                else if (!item.EnabledIsChecked)
+                {
+                    var dependencies = item.Dependencies
+                        .Select(x => _db.Items.First(i => i.Name == x))
+                        .Where(x => x.State is NotInstalledState).ToList();
+
+                    if (dependencies.Count == 0 || await DisplayErrors.DisplayHasNotInstalledDependenciesWarning(item.Name, dependencies))
+                    {
+                        foreach (var dependency in dependencies)
+                            await _installer.Install(dependency, _ => { }, true);
```

Comment:
if this throws error for whatever reason, app will crash cuz its not handled. Probably should call InternalModDownload to do the job cuz it handles the errors

Reply:
fixed